### ISSUE TITLE
TA for random steps when observed is NA

### ISSUE THREAD

**Issue by bsmity13** [jmsigner/amt#63]
Step 1 has no turn angle because there is no previous step (`ta_ = NA`). However, the random steps paired with step 1 do have a turn angle. Demonstration:

```
deer %>% 
  steps_by_burst() %>% 
  random_steps(n_control = 3) %>% 
  print(n = 9, width = 100)
```

Problem would arise for first step of any burst. Problem might be that `direction_p` should be `NA`? Isn't that direction of the previous step?

**Comment by jmsigner** [comment]
Yes that's true, I did note this a few weeks ago. When fitting an iSSF with `fit_issf()` this is not a problem because `survival::clogit()` removes these strata. I did add a function `remove_incomplete_strata()` to deal with such cases.

If `direction_p` was `NA`, then we would lose these steps and in cases we are only interested in `sl_` it doesn't matter. What do you think?

**Comment by mjones029** [comment]
Hi! Commenting here as this issue is still marked as open. 
I've been updating code originally written with amt v0.1.7 to current, and am finding a significant change in output arising from the `random_steps()` function. 
As an example: I have a bursted track with 587 steps and 25 bursts. In amt v0.1.7, when I draw 10 random steps for each observed step, I have a resulting dataset with 6,457 steps. This makes sense, as 587*11 = 6,457. I have 10 random steps for each observed step. 

With the current version of amt, running `random_steps()` on that same 587 steps produces only 6,171 steps. It seems like observed locations at the start of bursts - and their corresponding random steps - have been removed. I believe this results from the `remove_incomplete_strata()` function, as the first step in a burst always has turning angle `NA`. 

This doesn't seem like an appropriate data removal to me, as `direction_p` is not `NA`, but maybe I'm missing something obvious! This is a real challenge for reproducibility between old and new versions of amt, so I wanted to mention it in what seems to be an open issue.
